modules/programs/neovim.nix

```diff
@@ -111,6 +115,12 @@ in
         description = "The package to use for the neovim binary.";
       };
 
+      customizedPackage = mkOption {
```

Comment:
I suggest changing the name of this option to `finalPackage` to match the name in the `programs.emacs` module. Also, perhaps set it to `visible = false`?